abis_mapping/base.py

```diff
@@ -0,0 +1,45 @@
+"""Provides ABIS Mapper Base Class"""
+
+
+# Standard
+import abc
+
+# Third-Party
+import rdflib
+
+# Typing
+from typing import final
+
+
+class ABISMapper(abc.ABC):
+    """ABIS Mapper Base Class"""
+
+    # Concrete ABIS Mapper Registry
+    registry: dict[str, type["ABISMapper"]] = {}
+
+    @abc.abstractmethod
+    def apply_mapping(self, data: bytes) -> rdflib.Graph:
```

Comment:
I think the input data type should be at higher level than `bytes`, something like StringIO that support buffering and streaming with the decoding (UTF-8) done by the client.
If we decide to use pandas for our CSV reader, and I think we should, let's set the data type the same as `pandas.read_csv()` function: 
https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html
(`FilePath | ReadCsvBuffer[bytes] | ReadCsvBuffer[str]`)

Reply:
Added the type hinting as above in: eb060b8

Interestingly, I could not get my IDE to resolve the `pandas._typing.FilePath` / `pandas._typing.ReadCsvBuffer` imports, so not sure if this will be a problem.